libraries/resource_systemd_automount.rb

```diff
@@ -0,0 +1,21 @@
+require_relative 'resource_systemd_unit'
+require_relative 'systemd_automount'
+
+class Chef::Resource
+  class SystemdAutomount < Chef::Resource::SystemdUnit
+    self.resource_name = :systemd_automount
+    provides :systemd_automount
+
+    def unit_type(_ = nil)
```

Comment:
do we need this as a method override? unit resource can provide a class method to do this,
```ruby
def self.unit_type(type)
@@unit_type = type
end
```
and in in child resources..
```ruby
unit_type :automount
```